### PR TITLE
Bratseth/freeze other zones while suspending

### PR DESCRIPTION
**Pull request by bratseth**
@jonmv please review, recommend looking at the commits one by one

### CODE REVIEW

**Comment by bratseth** [comment]
I'll merge once you approve.

File: configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java
Comment (jonmv) [inline configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: I'd also check if you can resume, and then see that the application is resumed.

Comment (jonmv) [reply configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: Well ... nevermind. It's just backed by a mock, huh?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java]: Is this a breaking comment, or a broken github?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: Perhaps 
```java
return application.deployments().keySet().stream()
                  .anyMatch(other ->    ! other.equals(zone)
                                     &&   controller.applications().isSuspended(new DeploymentId(application.id(), other)));
```

**Review by jonmv — APPROVED** [review]
Looks great! :)

File: configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java
Comment (bratseth) [reply configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: Yep.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java
Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java]: The former - fixed.

**Comment by bratseth** [comment]
Hey CLA

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: No objection!

**Comment by mpolden** [comment]
Appears to have broken CD. Computing ready jobs fails with:
```
controller-cd-3:[2018-10-24 07:08:03.227] WARNING : configserver     Container.com.yahoo.vespa.hosted.controller.maintenance.Maintainer
controller-cd-3:        ReadyJobsTrigger failed. Will retry in 0 minutes
controller-cd-3:        exception=
controller-cd-3:        javax.ws.rs.NotFoundException: HTTP 404 Not Found
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:1020)
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:883)
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation.access$800(JerseyInvocation.java:92)
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation$3.call(JerseyInvocation.java:724)
controller-cd-3:        at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
controller-cd-3:        at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
controller-cd-3:        at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
controller-cd-3:        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:444)
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:720)
controller-cd-3:        at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:430)
controller-cd-3:        at org.glassfish.jersey.client.proxy.WebResourceFactory.invoke(WebResourceFactory.java:345)
controller-cd-3:        at com.sun.proxy.$Proxy95.isSuspended(Unknown Source)
controller-cd-3:        at com.yahoo.vespa.hosted.clients.configserver.ConfigServerClientImpl.lambda$isSuspended$1(ConfigServerClientImpl.java:128)
controller-cd-3:        at com.yahoo.vespa.hosted.clients.configserver.ConfigServerClientImpl.retryOperationWithVip(ConfigServerClientImpl.java:423)
controller-cd-3:        at com.yahoo.vespa.hosted.clients.configserver.ConfigServerClientImpl.retryOperation(ConfigServerClientImpl.java:380)
controller-cd-3:        at com.yahoo.vespa.hosted.clients.configserver.ConfigServerClientImpl.isSuspended(ConfigServerClientImpl.java:125)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.ApplicationController.isSuspended(ApplicationController.java:619)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.isSuspendedInAnotherZone(DeploymentTrigger.java:367)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.canTrigger(DeploymentTrigger.java:354)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.lambda$computeReadyJobs$19(DeploymentTrigger.java:311)
controller-cd-3:        at java.util.Optional.ifPresent(Optional.java:159)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.computeReadyJobs(DeploymentTrigger.java:292)
controller-cd-3:        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
controller-cd-3:        at java.util.Iterator.forEachRemaining(Iterator.java:116)
controller-cd-3:        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
controller-cd-3:        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
controller-cd-3:        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
controller-cd-3:        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
controller-cd-3:        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
controller-cd-3:        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.computeReadyJobs(DeploymentTrigger.java:284)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.deployment.DeploymentTrigger.triggerReadyJobs(DeploymentTrigger.java:153)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.maintenance.ReadyJobsTrigger.maintain(ReadyJobsTrigger.java:21)
controller-cd-3:        at com.yahoo.vespa.hosted.controller.maintenance.Maintainer.run(Maintainer.java:62)
controller-cd-3:        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
controller-cd-3:        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
controller-cd-3:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
controller-cd-3:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
controller-cd-3:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
controller-cd-3:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
controller-cd-3:        at java.lang.Thread.run(Thread.java:748)
```

**Comment by jonmv** [comment]
Oh, it's _still_ broken ...

**Pull request merged**
